### PR TITLE
Added Order model

### PR DESCRIPTION
**Pull request by YaroslavRadevychVynnytskyi**

### CODE REVIEW

File: src/main/java/application/bookstore/model/Order.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                 String shippingAddress,
+                 Set<OrderItem> orderItems) {
+        this.id = id;
+        this.user = user;
+        this.status = status;
+        this.total = total;
+        this.orderDate = orderDate;
+        this.shippingAddress = shippingAddress;
+        this.orderItems = orderItems;
+    }
```
Comment (MentorMA) [inline src/main/java/application/bookstore/model/Order.java]: Tou already have builder and lombok annotations, so you don`t need this anymore

**Review by MentorMA — CHANGES_REQUESTED** [review]

File: src/main/java/application/bookstore/service/order/OrderService.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package application.bookstore.service.order;
+
+import application.bookstore.dto.OrderPlacementRequestDto;
+import application.bookstore.dto.order.OrderDto;
+import application.bookstore.dto.order.UpdateOrderStatusRequestDto;
+import application.bookstore.dto.orderitem.OrderItemDto;
+import java.util.List;
+
+public interface OrderService {
+    OrderDto placeOrderByUserId(Long id, OrderPlacementRequestDto request);
```
Comment (Rommelua) [inline src/main/java/application/bookstore/service/order/OrderService.java]:
```suggestion
    OrderDto placeOrderByUserId(Long userId, OrderPlacementRequestDto request);
```

File: src/main/java/application/bookstore/service/order/OrderService.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    List<OrderDto> getOrders(Long id);
```
Comment (Rommelua) [inline src/main/java/application/bookstore/service/order/OrderService.java]: userId?

File: src/main/java/application/bookstore/service/order/impl/OrderServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto placeOrderByUserId(Long id, OrderPlacementRequestDto request) {
+        User user = userRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Can't get user by id: " + id));
+        Order order = buildOrder(user, request.shippingAddress());
+        Order savedOrder = orderRepository.save(order);
+        orderItemRepository.saveAll(savedOrder.getOrderItems());
```
Comment (Rommelua) [inline src/main/java/application/bookstore/service/order/impl/OrderServiceImpl.java]: why don't you use cascade for saving items?

File: src/main/java/application/bookstore/service/order/impl/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    private final ShoppingCartRepository cartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto placeOrderByUserId(Long id, OrderPlacementRequestDto request) {
```
Comment (Rommelua) [inline src/main/java/application/bookstore/service/order/impl/OrderServiceImpl.java]: userId

File: src/main/java/application/bookstore/dto/order/UpdateOrderStatusRequestDto.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package application.bookstore.dto.order;
+
+public record UpdateOrderStatusRequestDto(String status) {
```
Comment (Rommelua) [inline src/main/java/application/bookstore/dto/order/UpdateOrderStatusRequestDto.java]: Why String status? Use enum

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/application/bookstore/dto/order/UpdateOrderStatusRequestDto.java [hunk 7]
```diff
@@ -0,0 +1,6 @@
+package application.bookstore.dto.order;
+
+import application.bookstore.model.Order;
+
+public record UpdateOrderStatusRequestDto(Order.Status status) {
```
Comment (Rommelua) [inline src/main/java/application/bookstore/dto/order/UpdateOrderStatusRequestDto.java]: NotNull

File: src/main/java/application/bookstore/service/order/impl/OrderServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+
+    @Transactional
+    @Override
+    public OrderDto placeOrderByUserId(Long userId, OrderPlacementRequestDto request) {
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException("Can't get user by id: " + userId));
+        Order order = buildOrder(user, request.shippingAddress());
+        Order savedOrder = orderRepository.save(order);
```
Comment (Rommelua) [inline src/main/java/application/bookstore/service/order/impl/OrderServiceImpl.java]:
```suggestion
        order = orderRepository.save(order);
```

**Pull request merged**